code/modules/surgery/surgery.dm

```diff
@@ -1,173 +1,232 @@
 ///Datum Surgery Helpers//
 /datum/surgery
-	var/name
-	var/status = 1
-	var/list/steps = list()
-	/*
-	var/eyes	=	0
-	var/face	=	0
-	var/appendix =	0
-	var/ribcage =	0
-	var/head_reattach = 0									//Steps in a surgery
-	*/
-
-	var/can_cancel = 1
-	var/step_in_progress = 0
-	var/list/in_progress = list()									//Actively performing a Surgery
-	var/location = "chest"										//Surgery location
-	var/requires_organic_bodypart = 1							//Prevents you from performing an operation on robotic limbs
-	var/list/possible_locs = list() 							//Multiple locations -- c0
-	var/obj/item/organ/organ_ref									//Operable body part
-	var/current_organ = "organ"
-	var/list/allowed_mob = list(/mob/living/carbon/human)
-
-/datum/surgery/proc/can_start(mob/user, mob/living/carbon/target)
-	// if 0 surgery wont show up in list
-	// put special restrictions here
-	return 1
-
-
-/datum/surgery/proc/next_step(mob/user, mob/living/carbon/target)
-	if(step_in_progress)	return
-
-	var/datum/surgery_step/S = get_surgery_step()
+	var/current_stage = SURGERY_STAGE_START
+	var/step_in_progress = FALSE
```

Comment:
Gonna refrain from adding that now. Adding a reference to a mob is something you have to think about since having a reference means that you also need to dereference it. Else the referenced mob won't properly GC.
Would be something for the operation computer improvement though